crates/shaders/src/compile/mod.rs

```diff
@@ -18,8 +18,12 @@ use crate::types::{BindType, BindingInfo, WorkgroupBufferInfo};
 
 #[derive(Error, Debug)]
 pub enum Error {
-    #[error("failed to parse shader: {0}")]
-    Parse(#[from] wgsl::ParseError),
+    #[error("failed to parse shader ({name}): {msg}")]
+    Parse {
+        name: String,
+        msg: String,
+        error: wgsl::ParseError,
+    },
 
     #[error("failed to validate shader: {0}")]
     Validate(#[from] WithSpan<ValidationError>),
```

Comment:
The error message for validation errors is unfortunately terrible, though. Again, not really a job for this PR

Reply:
Done. This will now result in an error like this on validation:

```shell
Failed to create renderer: Failed to compile shader: failed to validate shader (flatten): error: Function [30] 'es_seg_flatten_evolute' is invalid
    ┌─ wgsl:673:1
    │
673 │ ╭ fn es_seg_flatten_evolute(
674 │ │     es: EulerSeg,
675 │ │     path_ix: u32,
676 │ │     transform: Transform,
677 │ │     chord_len: f32,
678 │ │     tol: f32,
679 │ │ ) -> vec2f {
    │ ╰────────────^ naga::Function [30]
    │
    = The `return` value None does not match the function return value
```